woocommerce/woocommerce-gutenberg-products-block
# Migrate Storybook to 6.4.0 and enable auto-detection of controls (#5249)

Previously our Babel configuration conflicted with the one provided by default by Storybook, making us unable to leverage the automatic detection of properties from both JS and TS components to create controls (formerly known as knobs).

Upgrading to 6.4.0 fixes a bug that makes it easier to work with Babel configurations. Also done here:

* Removed a few deprecated addons in favor of the replacement ones bundled within Storybook Essentials.
* Add auto-detection of certain control types depending on the property names
* Add support for TS stories

More context: https://github.com/storybookjs/storybook/issues/12292

Fixes #5249

### Screenshots

![Screen Shot 2021-11-26 at 20 21 51](https://user-images.githubusercontent.com/1847066/143623236-223f9e0e-c416-4238-912a-c7b6aaff8107.png)

### Testing

#### Manual Testing

How to test the changes in this Pull Request:

1. Run `npm run storybook`
2. …be patient 😬 
3. Open Storybook at localhost:6006, in particular, you can look at [this component](http://localhost:6006/?path=/story/woocommerce-blocks-base-components-priceslider--default).
4. Notice there is a new `Controls` tab, where the properties of the components are auto-detected and documented along with the description

> ⚠️ &nbsp;**Please note:** This PR is only addressing the migration and the detection and not touching any stories. As such, I am temporarily keeping the legacy knobs, until each story is properly migrated to the controls paradigm.

### User Facing Testing

N/A (internal)

### Changelog

> Migrated to Storybook 6.4.0 with auto-detection of Controls
